src/com/xilinx/rapidwright/interchange/MacroParamRule.java

```diff
@@ -0,0 +1,68 @@
+package com.xilinx.rapidwright.interchange;
+
+import com.xilinx.rapidwright.design.Unisim;
+
+public class MacroParamRule {
+    public String primParam;
+    public String instName;
+    public String instParam;
+    public int[] bitSlice;
+    public MacroParamTableEntry[] tableLookup;
+
+    public MacroParamRule(String primParam, String instName, String instParam) {
+        this.primParam = primParam;
+        this.instName = instName;
+        this.instParam = instParam;
+        this.bitSlice = null;
+        this.tableLookup = null;
+    }
+
+    public static MacroParamRule copy(String primParam, String instName, String instParam) {
+        return new MacroParamRule(primParam, instName, instParam);
+    }
+
+    public static MacroParamRule bit_range(String primParam, String instName, String instParam, int start, int count) {
```

Comment:
Would it be ok to switch to camelCase? (`bit_range() --> bitRange()`)

Reply:
sure, happy to fix if we proceed with this